core/src/test/java/kafka/test/junit/ClusterTestExtensions.java

```diff
@@ -179,8 +186,8 @@ private void processClusterTest(ExtensionContext context, ClusterTest annot, Clu
                 throw new IllegalStateException();
         }
 
-        ClusterConfig.Builder builder = ClusterConfig.clusterBuilder(type, brokers, controllers, autoStart,
-            annot.securityProtocol(), annot.metadataVersion());
+        ClusterConfig.Builder builder = ClusterConfig.clusterBuilder(type, brokers, controllers, disksPerBroker,
```

Comment:
It seems to me `clusterBuilder` should be removed. Also, the temporary local variables (`type`, `brokers`, etc ) can be removed. We have a builder already and hence we should leverage it to collect all variables. for example:
```java
        ClusterConfig.Builder builder = ClusterConfig.builder();
        if (annot.clusterType() == Type.DEFAULT) {
            builder.type(defaults.clusterType());
        } else {
            builder.type(annot.clusterType());
        }
```

Reply:
Do we want to use another jira to trace this? Or I remove it in this PR? Thanks.

Reply:
We can do that in this PR as most changes involved by refactor are in this method. Let's do it to decorate our test infrastructure:)

Reply:
Thanks for the suggestion. I updated it.